desc/compute/_field.py

```diff
@@ -140,6 +141,7 @@ def _B(params, transforms, profiles, data, **kwargs):
     data=["B"],
 )
 def _B_R(params, transforms, profiles, data, **kwargs):
+    errorif(kwargs.get("basis", "rpz").lower() != "rpz", NotImplementedError)
```

Comment:
With #1027, `B` will be in `rpz` by default during compute functions, so as long as every compute function assumes `rpz` it is consistent. Then, 3D quantities will converted "after" actual calculation. What you did would be relevant if we had `B_X` and it was defined as `data["B"][:, 0]` without any prior conversion.